### PR TITLE
Add MacOS tool in assemble.sh

### PR DESCRIPTION
**Pull request by duc-nx**
Add MacOS tool in assemble.sh

Most of us use Mac.
I suggest change the command to native mactool.

Update:

This script will now detect if MacOS or not and use the tool accordingly.

Summary:

Test Plan:

### CODE REVIEW

**Review by yoichi-nexus — APPROVED** [review]
Looks good.

**Review by yoichi-nexus — DISMISSED** [review]
Looks good.

File: runtime/assemble.sh [hunk 1]
```diff
@@ -6,8 +6,8 @@ rm -f bin/*.a
 
 for ext in i imc
 do
-    riscv-none-elf-gcc -c -mabi=ilp32 -march=rv32${ext} -mcmodel=medlow asm.S -o bin/nexus-rt.o
-    riscv-none-elf-ar crs bin/riscv32${ext}-unknown-none-elf.a bin/nexus-rt.o
+    riscv64-elf-gcc -c -mabi=ilp32 -march=rv32${ext} -mcmodel=medlow asm.S -o bin/nexus-rt.o
```
Comment (sjudson) [inline runtime/assemble.sh:9]: I would prefer we either don't change these, or try to detect the OS and swap between them for Linux/MacOS.

**Review by sjudson — CHANGES_REQUESTED** [review]

File: runtime/assemble.sh (hunk 1 above)
Comment (duc-nx) [reply runtime/assemble.sh]: Good idea. Why didn't I think of it. Updated in c6f7716

File: runtime/assemble.sh [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 for ext in i imc
 do
-    riscv-none-elf-gcc -c -mabi=ilp32 -march=rv32${ext} -mcmodel=medlow asm.S -o bin/nexus-rt.o
-    riscv-none-elf-ar crs bin/riscv32${ext}-unknown-none-elf.a bin/nexus-rt.o
+    ${GCC_PREFIX}gcc -c -mabi=ilp32 -march=rv32${ext} -mcmodel=medlow asm.S -o bin/nexus-rt.o
+    ${GCC_PREFIX}ar crs bin/riscv32${ext}-unknown-none-elf.a bin/nexus-rt.o
 done
-
-rm bin/nexus-rt.o
+rm bin/nexus-rt.o
```
[the remark this answers is not in the document]
Comment (cothan) [reply runtime/assemble.sh]: done in https://github.com/nexus-xyz/nexus-zkvm/commit/6c5d989f954fd214d68f2df8bbbc1be06ef01fb8

[the remark this answers is not in the document]
Comment (duc-nx) [reply runtime/assemble.sh]: done in https://github.com/nexus-xyz/nexus-zkvm/commit/6c5d989f954fd214d68f2df8bbbc1be06ef01fb8

**Review by sjudson — APPROVED** [review]

**Pull request merged**